File: app/src/main/java/com/coinninja/coinkeeper/model/db/enums/Type.java
package com.coinninja.coinkeeper.model.db.enums;

public enum Type {
    SENT(0),
    RECEIVED(10);

    private final int id;

    Type(int id) {
        this.id = id;
    }

    public int getId() {
        return id;
    }
}

File: app/src/main/java/com/coinninja/coinkeeper/model/db/enums/Type.kt
package com.coinninja.coinkeeper.model.db.enums

enum class Type(val id: Int) {
    BLOCKCHAIN_SENT(0),
    BLOCKCHAIN_RECEIVED(10),
    LIGHTNING_SENT(20),
    LIGHTNING_RECEIVED(30);

    companion object {
        fun sentFrom(value: String) = when (value) {
            "lightning" -> LIGHTNING_SENT
            else -> BLOCKCHAIN_SENT
        }

        fun receivedFrom(value: String): Type = when (value) {
            "lightning" -> LIGHTNING_RECEIVED
            else -> BLOCKCHAIN_RECEIVED
        }
    }

}